### PR TITLE
Abstract file access for tensor as external resources

### PR DESCRIPTION
**Pull request by kali**

### CODE REVIEW

File: onnx/src/data_resolver.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+pub fn default() -> Box<dyn ModelDataResolver> {
+    Box::new(MmapDataResolver)
+}
+
+#[cfg(target_family = "wasm")]
+pub fn default() -> Box<dyn ModelDataResolver> {
+    Box::new(FopenDataResolver)
+}
+
+pub trait ModelDataResolver {
```
Comment (kckeiks) [inline onnx/src/data_resolver.rs:18]: This should work @kali 

Not too important but I wonder if these methods can be consolidated to make the trait simpler.

Comment (kali) [reply onnx/src/data_resolver.rs:18]: I'm not sure we can without having it lead to extra copies here and there. Can you see a way ?

Comment (kckeiks) [reply onnx/src/data_resolver.rs:18]: Hmm where is `read()` used? Couldn't we just have `read_bytes_from_path`?

Comment (kali) [reply onnx/src/data_resolver.rs:18]: indeed. not sure why I incuded it in the first place.

**Pull request merged**